micro_http/src/request.rs

```diff
@@ -22,16 +22,18 @@ fn find(bytes: &[u8], sequence: &[u8]) -> Option<usize> {
 ///
 /// The `Uri` can not be used directly and it is only accessible from an HTTP Request.
 #[derive(Clone, Debug, PartialEq)]
-pub struct Uri<'a> {
-    slice: &'a str,
+pub struct Uri {
```

Comment:
This change is not necessarily relevant to this commit, you should create a new commit for this change and for all the other changes that aren't related to adding the connection.

Reply:
Since you are putting this in a different commit, you should also specify **why** you are changing this from str to a String.